extensions/azure-keyvault/runtime/src/main/java/io/quarkiverse/azure/keyvault/secret/runtime/KeyVaultSecretClientProducer.java

```diff
@@ -18,29 +19,29 @@ public class KeyVaultSecretClientProducer {
 
     @Produces
     public SecretClient createSecretClient() {
-        if (!secretConfiguration.enabled) {
+        if (!secretConfiguration.enabled()) {
             return null;
         }
 
-        assert secretConfiguration.endpoint.isPresent() : "The endpoint of Azure Key Vault Secret must be set";
+        assert secretConfiguration.endpoint().isPresent() : "The endpoint of Azure Key Vault Secret must be set";
         return new SecretClientBuilder()
                 .clientOptions(new ClientOptions().setApplicationId(AzureQuarkusIdentifier.AZURE_QUARKUS_KEY_VAULT_SYNC_CLIENT))
-                .vaultUrl(secretConfiguration.endpoint.get())
+                .vaultUrl(secretConfiguration.endpoint().get())
                 .credential(new DefaultAzureCredentialBuilder().build())
                 .buildClient();
     }
 
     @Produces
     public SecretAsyncClient createSecretAsyncClient() {
-        if (!secretConfiguration.enabled) {
+        if (!secretConfiguration.enabled()) {
             return null;
         }
 
-        assert secretConfiguration.endpoint.isPresent() : "The endpoint of Azure Key Vault Secret must be set";
+        assert secretConfiguration.endpoint().isPresent() : "The endpoint of Azure Key Vault Secret must be set";
```

Comment:
Now we support multiple key vaults, maybe `endpoint` can be optional and return null if it's not provided?

Reply:
Duplicated comment, see above reply.